pkg/daemon/update.go

```diff
@@ -732,7 +979,48 @@ func (dn *Daemon) update(oldConfig, newConfig *mcfgv1.MachineConfig, skipCertifi
 	logSystem("Starting update from %s to %s: %+v", oldConfigName, newConfigName, diff)
 
 	diffFileSet := ctrlcommon.CalculateConfigFileDiffs(&oldIgnConfig, &newIgnConfig)
-	actions, err := calculatePostConfigChangeAction(diff, diffFileSet)
+	diffUnitSet := ctrlcommon.CalculateConfigUnitDiffs(&oldIgnConfig, &newIgnConfig)
+
+	var fg featuregates.FeatureGate
+
+	// This check is needed as featureGatesAccessor is not present during first boot. During firstboot
+	// the daemon will always do a reboot and NodeDisruptionPolicies are not active.
+	if dn.featureGatesAccessor != nil {
+		fg, err = dn.featureGatesAccessor.CurrentFeatureGates()
+		if err != nil {
+			klog.Errorf("Could not get fg: %v", err)
+			return err
+		}
+	}
+
+	var nodeDisruptionActions []opv1.NodeDisruptionPolicyStatusAction
+	var actions []string
+	// If FeatureGateNodeDisruptionPolicy is set, calculate NodeDisruptionPolicy based actions for this MC diff
+	if fg != nil && fg.Enabled(v1.FeatureGateNodeDisruptionPolicy) {
+		mcop, err := dn.mcopClient.OperatorV1().MachineConfigurations().Get(context.TODO(), "cluster", metav1.GetOptions{})
+		if mcop == nil || apierrors.IsNotFound(err) {
+			return err
+		}
+		// TODO: wait for mcop.Status.NodeDisruptionPolicyStatus to populate, otherwise, error out?
```

Comment:
I suppose its fine to have a check for whether clusterpolicies is empty, since it shouldn't if the MCO has populated it (maybe fallback?)

Reply:
Yeah I wonder if there is a time window where clusterpolicies aren't ready yet, but the feature gate is on. We wouldn't want to immediately fall back in that case, so perhaps a timeout? I haven't seen this in my testing, it'd have to be a very small edge case